src/jit/codegenxarch.cpp

```diff
@@ -1329,9 +1345,23 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
     }
     else
     {
-        // dividend must be in RAX
-        if (dividend->gtRegNum != REG_RAX)
+        if (dividendIsLong)
+        {
+            assert(dividendLo != nullptr && dividendHi != nullptr);
+
+            // dividendLo must be in RAX; dividendHi must be in RDX
+            if (dividendLo->gtRegNum != REG_RAX)
```

Comment:
I'm a bit surprised that this is necessary (ditto the code below for non-long-typed dividends). Why doesn't the RA take care of this?

Also, is this safe w.r.t. GC info? E.g. if there was a GC reference in EAX will it be killed by this instruction, or will it be killed by the `div`? In the latter case, it seems like we'd have a value being reported as a GC ref that is not a valid GC ref. Is this okay?